aten/src/ATen/test/reduce_ops_test.cpp

```diff
@@ -8,9 +8,18 @@ using namespace at;
 TEST(ReduceOpsTest, MaxValuesAndMinValues) {
   const int W = 10;
   const int H = 10;
-  if (hasCUDA()) {
-    for (const auto dtype : {kHalf, kFloat, kDouble, kShort, kInt, kLong}) {
-      auto a = at::rand({H, W}, TensorOptions(kCUDA).dtype(at::kHalf));
+
+  for (const auto dev : {kCPU, kCUDA}) {
+    std::vector<at::ScalarType> dtypes = {kFloat, kDouble, kShort, kInt, kLong};
+    if (hasCUDA()) {
+      dtypes.push_back(kHalf);
+    } else if (dev == kCUDA) {
+      // Skip CUDA test in non cuda env
+      continue;
+    }
+
+    for (const auto dtype : dtypes) {
+      auto a = (at::rand({H, W}, TensorOptions(dev).dtype(kFloat)) * 10).to(dtype);
```

Comment:
You can initialize with randint and specify the necessary dtype, that way you won't have to do this awkward multiplication and cast. Also, if you have both positive and negative values in your tensor, you don't need additional tests with .neg